aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -599,6 +599,98 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void pre_check_gradient(const Tensor& self, IntArrayRef dim, int64_t spacing_size, int64_t edge_order, bool spacing_existence, bool dim_existence) {
+  // Helper for gradient function to make sure input data satisfies prerequisites
+  TORCH_CHECK(self.scalar_type() != ScalarType::Byte, "torch.gradient does not support uint8 input.");
+  if (spacing_existence && !dim_existence) {
+    TORCH_CHECK( spacing_size == 1 || spacing_size == self.dim(), "torch.gradent expected spacing has size of 1 or ", self.dim(), " or not given at all when dim is not provided." );
```

Comment:
torch.gradient expected spacing to be unspecified, a scalar or a list of length self.dim(), but got a list of length ####

Reply:
""""but got"""" is very consistently used in PyTorch